### PR TITLE
Added script to claim prizes and send email based on the teacher survey

### PR DESCRIPTION
**Pull request by aoby**
CC @jeremydstone because it's email related

### CODE REVIEW

File: bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+require 'set'
+
+# Reads responses from an exported form_responses.csv (not checked into the repo because it contains PII)
+# For responses that requested a prize, claim a prize and prepare to send email with that prize code.
+
+PRIZE_PURPOSE = 'TeacherImplementation1617'
+PRIZE_TYPE = 'Apple.US'
+
+COL_RECEIVE_PRIZE = 'Would you like to receive a $10 Apple gift card for completing this survey? If so, you will receive a follow up email by November 20th with instructions about how to claim your gift.'
+COL_EMAIL = 'Please provide your email address to receive your gift card.'
```
Comment (ashercodeorg) [inline bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb]: nit: Use `.freeze` on string constants.

File: bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  csv << [:email, :prize_code_s]
+  CSV.foreach('./form_responses.csv', headers: true) do |row|
+    next unless row[COL_RECEIVE_PRIZE] && row[COL_EMAIL]
+    next unless row[COL_RECEIVE_PRIZE].strip.downcase == 'yes'
+    email = row[COL_EMAIL].strip.downcase
+    next if email.blank? || emails.include?(email)
+    emails.add email
+
+    # Note claim_prize_code is idempotent.
+    # If this is called again with the same parameters, it will return the already claimed prize code.
```
Comment (ashercodeorg) [inline bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb:30]: Given that emails was declared as a set, why does idempotency matter?

File: bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+require_relative '../mailing-common/mailing-list-utils'
+require_relative '../../../pegasus/helpers/prize_helpers'
+require 'active_support'
+require 'active_support/core_ext/object/blank'
+require 'set'
+
+# Reads responses from an exported form_responses.csv (not checked into the repo because it contains PII)
+# For responses that requested a prize, claim a prize and prepare to send email with that prize code.
+
+PRIZE_PURPOSE = 'TeacherImplementation1617'
```
Comment (ashercodeorg) [inline bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb]: Is there meaning to the `1617` suffix? Assuming so, a comment seems helpful.

File: bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb (hunk 2 above)
Comment (ashercodeorg) [reply bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb:30]: Looking at the [code](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/forms/hoc_survey_2014.rb#L185), it returns the first claimed prize associated with `email`, not the first claimed prize associated with `email` and `PRIZE_TYPE`. Should this be changed?

Comment (ashercodeorg) [reply bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb:30]: Oops, wrong method, right `claim_prize_code` method [here](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/helpers/prize_helpers.rb#L7). LGTM.

File: bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb (hunk 3 above)
Comment (aoby) [reply bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb]: It's the 2016-2017 school year. I'll add a comment.

File: bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb (hunk 2 above)
Comment (aoby) [reply bin/mail/2016-11-10-teacher-survey-apple/claim_prizes_and_generate_list.rb:30]: Idempotency would only matter here if the script is run twice (which it shouldn't be). It's just a safeguard.

**Pull request merged**